projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -67,6 +67,16 @@ export class IgxFilteringService implements OnDestroy {
         const filterIcon = column.filteringExpressionsTree ? 'igx-excel-filter__icon--filtered' : 'igx-excel-filter__icon';
         const filterIconTarget = element.querySelector(`.${filterIcon}`) as HTMLElement || element;
 
+        this._overlayService.contentAppended
+            .pipe(
+                first(),
```

Comment:
Probably nearly impossible to break this, but the service being root singleton and with some concurency it is possible to pick up another overlay's event, so would be much happier with this checing like the other events below:
```suggestion
                first(overlay => overlay.id === id),
```

Reply:
Although, kind of interested if we can skip this extra event entirely or if the size adjust can be done either in order directly or during opening

Reply:
Removed the `contentAppended` as now there is not need to adjust the size